### PR TITLE
Remove internal APIs

### LINKED ISSUE #10: Cleanup: remove Internal namespaces from API reference
_From @divega on June 5, 2017 22:18_

E.g. https://docs.microsoft.com/en-us/ef/core/api/microsoft.entityframeworkcore.design.internal. 

_Copied from original issue: aspnet/EntityFramework.Docs#402_

### PR DESCRIPTION
**Pull request by mairaw**
Fixes #10

### CODE REVIEW

**Comment by mairaw** [comment]
It seems that a couple of APIs link to some of the internal APIs like https://review.docs.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.storage.relationaldatabaseproviderservices.expressionprinter?view=efcore-1.1&branch=pr-en-us-59.

In this case, the public API inherits from the internal. Do you want me to exclude the internal APIs that are being linked to from the filter and keep them on docs?

**Comment by smitpatel** [comment]
@mairaw - In past we had API mess where things depended on Internal namespace where it shouldn't have been. We have cleaned up a lot in 3.0 release. Let me discuss with team what do we want to do with that regard in previous versions. and get back to you tomorrow.

File: dotnet/xml/_filter.xml [hunk 1]
```diff
@@ -1,8 +1,170 @@
 <?xml version="1.0" encoding="utf-8"?>
 <filter>
   <apiFilter apply="true">
+    <namespaceFilter name="Microsoft.EntityFrameworkCore.ChangeTracking.Internal">
```
Comment (smitpatel) [inline dotnet/xml/_filter.xml]: Would this list be, all namespace across all versions which should be filtered out?

Comment (mairaw) [reply dotnet/xml/_filter.xml]: yes

**Comment by smitpatel** [comment]
@mairaw - I think what we have in this PR is good. public property in public class should be in docs but reference to .Internal type should be not present. A text rather than a link. I checked the link you shared that seems perfect thing we want to have. Thanks.

File: dotnet/xml/_filter.xml (hunk 1 above)
Comment (mairaw) [reply dotnet/xml/_filter.xml]: Yes.

**Comment by mairaw** [comment]
Great. I'll merge this then. You'll just have a couple of warnings then in the build report that you can ignore then.

**Pull request merged**